### ISSUE TITLE
build error in xodr.h

### ISSUE THREAD

**Issue by breitlauch** [javedulu/ad-xolib#14]
First things first: Thank you for sharing this useful library! :)

Unfortunately I'm running into a build error, no matter if I use the header files directly in my C++ project or if I build the project via cmake.

I got the following error messages: 
`xodr.h(30): error C2971: '
t_patternstr': template parameter 'pattern': 'e_countryCode_iso3166alpha2_pattern': a variable with non-static storage
duration cannot be used as a non-type argument`

`\xosc.h(27): error C2971: '
t_patternstr': template parameter 'pattern': 'parameter_pattern': a variable with non-static storage duration cannot be
 used as a non-type argument`

For my own C++ project I'm using Qt 5.14.1 MSVC2017 64bit with C++17 standard.
For the cmake build I used the following commands (--target was not working for me):
`cd 'path-to-OpenDriveScenarioParser'`
`mkdir build`
`cmake . -B build -G "Visual Studio 15 2017 Win64"`
`cmake --build build`

I'm very curious to know where the error comes from. Apparently it has not occurred so far?!

**Comment by javedulu** [comment]
I dont have a windows machine, could you please make this change and let me know if it works .

const constexpr char e_countryCode_iso3166alpha2_pattern[] = "[A-Z]{2}"; 

static const constexpr char e_countryCode_iso3166alpha2_pattern[] = "[A-Z]{2}"; 

add static before the const constexpr : if this works i can add it globally to all derived constexpr variables

**Comment by breitlauch** [comment]
I think, I already tried that before. However, this is not working. For some reason the `constexpr` values are not available at compile time.

**Comment by breitlauch** [comment]
Wow, I solved "the problem". I don't know why, but I must have forgotten to execute one of Qt's build in commands to clean the old compile files, to run qmake and rebuild. I assume, that especially the "clean" does the trick. After adding your header files to my project the compiler keeps telling me, that the variables are non-static - which is strange by using `constexpr`. Maybe some existing (old) files were not recreated, which is why the compiler had no chance to know the `constexpr` values.

So, long story short: I changed nothing, no `static` is needed. Just clean, run qmake and recompile. Hilarious!

Thank you for your fast support!
